src/main/java/org/zalando/nakadi/repository/db/EventTypeCache.java

```diff
@@ -3,63 +3,103 @@
 import com.google.common.cache.CacheBuilder;
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
-import org.zalando.nakadi.domain.EventType;
-import org.zalando.nakadi.exceptions.InternalNakadiException;
-import org.zalando.nakadi.exceptions.NoSuchEventTypeException;
-import org.zalando.nakadi.repository.EventTypeRepository;
-import org.zalando.nakadi.validation.EventTypeValidator;
-import org.zalando.nakadi.validation.EventValidation;
 import org.apache.curator.framework.CuratorFramework;
 import org.apache.curator.framework.recipes.cache.PathChildrenCache;
 import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
-import org.apache.curator.framework.recipes.cache.PathChildrenCacheListener;
 import org.apache.curator.utils.ZKPaths;
 import org.apache.zookeeper.CreateMode;
 import org.apache.zookeeper.KeeperException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.exceptions.InternalNakadiException;
+import org.zalando.nakadi.exceptions.NoSuchEventTypeException;
+import org.zalando.nakadi.repository.EventTypeRepository;
+import org.zalando.nakadi.validation.EventTypeValidator;
+import org.zalando.nakadi.validation.EventValidation;
 
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Optional;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
 
 public class EventTypeCache {
 
     public static final String ZKNODE_PATH = "/nakadi/event_types";
     public static final int CACHE_MAX_SIZE = 100000;
+    private static final Logger LOG = LoggerFactory.getLogger(EventTypeCache.class);
+
+    private static class CachedValue {
+        final EventType eventType;
```

Comment:
For me the difference is very big.
Currently your class is mutable. If you make these fields private - it will become immutable.

Actually I don't know a single good case to make a property with package visibility.